packages/openapi-typescript/src/types.ts

```diff
@@ -543,44 +543,54 @@ export interface XMLObject extends Extensable {
  * [4.8.27] Security Scheme Object
  * Defines a security scheme that can be used by the operations.
  */
-export type SecuritySchemeObject = {
-  /** A description for security scheme. CommonMark syntax MAY be used for rich text representation. */
+export interface SecuritySchemeObject {
+  /**
+   * REQUIRED.
+   * Applies To: ANY.
+   * The type of the security scheme. Valid values are "apiKey", "http", "oauth2", "openIdConnect".
+   */
+  type: "apiKey" | "http" | "oauth2" | "openIdConnect";
+  /**
+   * Applies To: ANY.
+   * A short description for security scheme. CommonMark syntax MAY be used for rich text representation.
+   */
   description?: string;
-  [key: `x-${string}`]: any;
-} & (
-  | {
-      /** REQUIRED. The type of the security scheme. */
-      type: "apiKey";
-      /** REQUIRED. The name of the header, query or cookie parameter to be used. */
-      name: string;
-      /** REQUIRED. The location of the API key. */
-      in: "query" | "header" | "cookie";
-    }
-  | {
-      /** REQUIRED. The type of the security scheme. */
-      type: "http";
-      /** REQUIRED. The name of the HTTP Authorization scheme to be used in the Authorization header as defined in [RFC7235]. The values used SHOULD be registered in the IANA Authentication Scheme registry. */
-      scheme: string;
-      /** A hint to the client to identify how the bearer token is formatted. Bearer tokens are usually generated by an authorization server, so this information is primarily for documentation purposes. */
-      bearer?: string;
-    }
-  | {
-      /** REQUIRED. The type of the security scheme. */
-      type: "mutualTLS";
-    }
-  | {
-      /** REQUIRED. Tye type of the security scheme. */
-      type: "oauth2";
-      /** REQUIRED. An object containing configuration information for the flow types supported. */
-      flows: OAuthFlowsObject;
-    }
-  | {
-      /** REQUIRED. Tye type of the security scheme. */
```

Comment:
Question: why this change here? My understanding was that other properties are required or not based on the `type`; this changes them so that they’re all potentially nullable, which feels like a regression?